File: app/src/main/java/com/github/brianspace/moviebrowser/di/modules/MainActivityModule.java
package com.github.brianspace.moviebrowser.di.modules;

import com.github.brianspace.moviebrowser.ui.activity.MainActivity;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Dagger module for MainActivity.
 */
@Module
public abstract class MainActivityModule {
    @ContributesAndroidInjector
    /* default */ abstract MainActivity contributeMainActivityInjector();
}

File: app/src/main/java/com/github/brianspace/moviebrowser/di/modules/MainActivityModule.kt
package com.github.brianspace.moviebrowser.di.modules

import com.github.brianspace.moviebrowser.ui.activity.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Dagger module for MainActivity.
 */
@Module
abstract class MainActivityModule {
    @ContributesAndroidInjector
    internal abstract fun contributeMainActivityInjector(): MainActivity
}
